tests/Cache/CacheClientTest.php

```diff
@@ -2333,4 +2336,118 @@ public function testSetDelete_SetExists_HappyPath()
     }
+
+    public function testGetBatch_HappyPath()
+    {
+        $cacheName = $this->TEST_CACHE_NAME;
+        $key1 = uniqid();
+        $key2 = uniqid();
+        $key3 = uniqid();
+        $keys = [$key1, $key2, $key3];
+
+        $value1 = uniqid();
+        $value2 = uniqid();
+        $value3 = uniqid();
+        $expectedValues = [$value1, $value2, $value3];
+
+        $response = $this->client->set($cacheName, $key1, $value1);
+        $this->assertNull($response->asError());
+        $this->assertNotNull($response->asSuccess(), "Expected a success but got: $response");
+        $this->assertEquals("$response", get_class($response));
+
+        $response = $this->client->set($cacheName, $key2, $value2);
+        $this->assertNull($response->asError());
+        $this->assertNotNull($response->asSuccess(), "Expected a success but got: $response");
+        $this->assertEquals("$response", get_class($response));
+
+        $response = $this->client->set($cacheName, $key3, $value3);
+        $this->assertNull($response->asError());
+        $this->assertNotNull($response->asSuccess(), "Expected a success but got: $response");
+        $this->assertEquals("$response", get_class($response));
+
+
+        $response = $this->client->getBatch($cacheName, $keys);
+
+        $this->assertNull($response->asError());
+        $this->assertNotNull($response->asSuccess(), "Expected a success but got: $response");
+
+        $responses = $response->asSuccess()->values();
+        $this->assertCount(sizeof($keys), $responses);
+        $this->assertEquals($responses, $expectedValues);
+    }
+
+    public function testSetBatch_HappyPath()
+    {
+        $cacheName = $this->TEST_CACHE_NAME;
+        $key1 = uniqid();
+        $key2 = uniqid();
+        $key3 = uniqid();
+        $keys = [$key1, $key2, $key3];
+
+        $value1 = uniqid();
+        $value2 = uniqid();
+        $value3 = uniqid();
+        $expectedValues = [$value1, $value2, $value3];
+
+        $items = array(
+            $key1 => $value1,
+            $key2 => $value2,
+            $key3 => $value3
+        );
+
+        $setBatchResponse = $this->client->setBatch($cacheName, $items, 60);
+        $this->assertNull($setBatchResponse->asError());
+        $this->assertNotNull($setBatchResponse->asSuccess(), "Expected a success but got: $setBatchResponse");
+        $setResponses = $setBatchResponse->asSuccess()->results();
+        $this->assertCount(sizeof($keys), $setResponses);
+
+        $getBatchResponse = $this->client->getBatch($cacheName, $keys);
+        $success = $getBatchResponse->asSuccess();
+        $this->assertNull($getBatchResponse->asError());
+        $this->assertNotNull($success, "Expected a success but got: $getBatchResponse");
+
+        $responses = $success->values();
+        $this->assertCount(sizeof($keys), $responses);
+        $this->assertEquals($responses, $expectedValues);
+    }
+
+    public function testGetBatchSetBatch()
```

Comment:
Could you please add a copy of this test that uses the async methods to provide an example of how those should be used? I have it on my to do list to the same for the scalar async methods.